backend/src/controllers/manager/assignEmployeeCourses.ts

```diff
@@ -0,0 +1,77 @@
+import { NextFunction, Request, Response } from 'express';
+
+import BadRequestError from 'classes/errors/clientErrors/BadRequestError';
+import { INITIAL_INDX } from 'config/constants';
+import {
+  assignCourseToEmployee,
+  getAllClientCoursesProvider,
+} from 'db/providers/clientCourseProvider';
+import { materialsCounterProvider } from 'db/providers/courseProvider';
+import { getUserProvider } from 'db/providers/userProvider';
+import { ICourseToAssign } from 'interfaces/ICourses/IQueryCourses';
+import { generateProgressDto } from 'utils/dto/dtoUtils';
+import { checkCourseDuplicates } from 'utils/validation/checkDuplicates';
+import { removeCoursesToAssignDuplicates } from 'utils/normaliser/queryCourses';
+
+const assignEmployeeCourses = async (
+  req: Request<{ id: string }, never, ICourseToAssign[]>,
+  res: Response<never, { id: string; results: string }>,
+  next: NextFunction,
+) => {
+  try {
+    const { id: employeeId } = req.params;
+    const { id: managerId } = res.locals;
+    const coursesToAssign = req.body;
+
+    const employee = await getUserProvider(employeeId);
+    const manager = await getUserProvider(managerId);
+
+    if (String(employee.managerId) !== String(manager._id)) {
+      throw new BadRequestError('Employee does not belong to user.');
+    }
+
+    const reducedCoursesToAssign = removeCoursesToAssignDuplicates(coursesToAssign);
+
+    const employeeClientCourses = await getAllClientCoursesProvider(employeeId);
+
+    const duplicates = reducedCoursesToAssign.reduce((foundDuplicates, courseToAssign) => {
```

Comment:
seems  like this fn has a lot in common with utils.queryCourses

Reply:
At least this function has two arguments unlike my util. And I don't think it's a good idea to make "god function" that will solve both issues. I can extract this lambda to another util function.